python/felis/cli.py

```diff
@@ -174,15 +167,35 @@ def load_tap(
         tap_key_columns_table,
     )
 
-    if engine_url == "sqlite://" and not dry_run:
-        # In Memory SQLite - Mostly used to test
-        Tap11Base.metadata.create_all(engine)
+    if not dry_run:
+        engine = create_engine(engine_url)
 
-    for schema in normalized["@graph"]:
-        tap_visitor = TapLoadingVisitor(
-            engine, catalog_name=catalog_name, schema_name=schema_name, mock=dry_run, tap_tables=tap_tables
-        )
-        tap_visitor.visit_schema(schema)
+        if engine_url == "sqlite://" and not dry_run:
+            # In Memory SQLite - Mostly used to test
+            Tap11Base.metadata.create_all(engine)
+
+        for schema in normalized["@graph"]:
+            tap_visitor = TapLoadingVisitor(
```

Comment:
I suppose we could store `TapLoadingVisitor` or `TapLoadingVisit.from_mock_connection` in a variable and then have this loop outside the `if`-blocks so it's not repeated code, but it's probably not worth it.

Reply:
I thought about moving it out of the loop, but it looks like there is some accumulated internal state in the visitor, so it may not be 100% safe to do it. Maybe something to cleanup later.